src/components/app/ServiceWorkerManager.js

```diff
@@ -9,29 +9,72 @@ import { assertExhaustiveCheck } from '../../utils/flow';
 
 import { getDataSource } from '../../selectors/url-state';
 import { getView } from '../../selectors/app';
+import { getSymbolicationStatus } from '../../selectors/profile';
 
 import type { ConnectedProps } from '../../utils/connect';
 import type { DataSource } from '../../types/actions';
-import type { Phase } from '../../types/state';
+import type { Phase, SymbolicationStatus } from '../../types/state';
 
 import './ServiceWorkerManager.css';
 
 type StateProps = {|
   +dataSource: DataSource,
   +phase: Phase,
+  +symbolicationStatus: SymbolicationStatus,
 |};
 type Props = ConnectedProps<{||}, StateProps, {||}>;
 
-type InstallStatus = 'pending' | 'ready' | 'idle';
+type InstallStatus = 'pending' | 'installing' | 'installed' | 'idle';
 type State = {|
   installStatus: InstallStatus,
   isNoticeDisplayed: boolean,
+  updatedWhileNotReady: boolean,
 |};
 
+/**
+ * This component is responsible for installing and updating the service worker,
+ * possibly displaying an information notice for the user to switch to the new
+ * service worker and reload the page whenever a new version is ready and we're
+ * in a state that would accept a reload.
+ *
+ * Here are some assumptions:
+ * - The browser checks for an update (in background) only when registering the
+ *   SW, when the webapp starts up. There are other cases where this happens but
+ *   this is more rare, especially for us where there's not much network traffic.
+ * - By the time we have a symbolicated profile displayed, everything that
+ *   should be downloaded is already downloaded.
+ *
+ * This means that we can force the new service worker at that time: this
+ * shouldn't disturb other open tabs that are probably fully loaded already, nor
+ * this current tab that is also fully loaded.
+ *
+ * If we didn't force the new service worker, then, to update, the user would
+ * have to close all tabs first. This is not very nice, especially that it's
+ * common that our users have several tabs open.
+ *
+ * We decided to force the new service worker only when the user clicks the
+ * reload button. Here is why:
+ * - As said above, it's common that our users have several tabs open.
+ * - Other tabs will likely be "fully loaded".
+ * - Therefore other tabs will trigger the update, even though the current tab
+ *   isn't ready.
+ * That's why the conscious user action is useful: 1/ this gives an "artisanal"
+ * timeout, which could allow the symbolication to finish, and 2/ the user knows
+ * something happens and is in control of it.
+ *
+ * But be wary that this breaks the F5 capability: the user will _have_ to
```

Comment:
This is a bit unfortunate that F5 doesn't work :-/ I'm not really sure on the mechanisms here to give more feedback though.

Reply:
Yeah, I tried very hard to make it work, but I couldn't find a satisfying mechanism. Maybe by compromising a bit on some rare use cases...